File: Integer/ZeroSumArray/ZeroSumSub.java
class ZeroSumSub {
	public: 
	vector<int> zeroSumSub(vector<int> nums) {
		vector<int> result;
		
		int curr_num = 0;
		vector<int> sum;
		for(int i = 0; i < nums.size(); ++i) {
			curr_num += nums[i];
			
			if( 0 == curr_num) {
				result.push_back(0);
				result.push_back(i);
				return result;
			}
			
			vector<int> iterator::iter = find(sum.begin(); sum.end(); curr_num);
			if(iter != sum.end()) {
				result.push_back(iter - sum.begin() + 1);
				result.push_back(i);
				return result;
			}
			sum.push_back(curr_num);
		}
		
		
		return result;
		
	}
};

File: Integer/ZeroSumArray/ZeroSumSub.cc
class ZeroSumArray {
	public int[] zeroSumSub(int[] nums) {
		List<int> result = new ArrayList<int>();
		
		int curr_num = 0;
		List<int> sum = new ArrayList<int>();
		
		for(int i = 0; i < nums.length(); ++i) {
			curr_num += nums[i];
			
			if(curr_num == 0) {
				result.add(0);
				result.add(i);
				return result;
			}
			
			Iterator iter = sum.iterator();
			while(iter.hasNext()) {
				if(iter == curr_num) {
					result.add(iter - sum.get(0) + 1);
					result.add(i);
					return result;
				}
			}
			sum.add(curr_num);
			
		}
		return result;
		
	}
}